llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp

```diff
@@ -1693,6 +1728,43 @@ SPIRVType *SPIRVGlobalRegistry::getOrCreateSPIRVArrayType(
 }
 
 SPIRVType *SPIRVGlobalRegistry::getOrCreateSPIRVPointerType(
+    Type *BaseType, MachineInstr &I, SPIRV::StorageClass::StorageClass SC) {
+  MachineIRBuilder MIRBuilder(I);
+  return getOrCreateSPIRVPointerType(BaseType, MIRBuilder, SC);
+}
+
+SPIRVType *SPIRVGlobalRegistry::getOrCreateSPIRVPointerType(
+    Type *BaseType, MachineIRBuilder &MIRBuilder,
```

Comment:
Those `Type*` can be marked `const`, which will allow you to remove the `const_cast`

Reply:
Good catch. I believe I did not add the const because the original function with the SPIRVType did not use const.

I still have to leave on const cast, but this is a little better.